### DISCUSSION TITLE
reporting all emissions

### DISCUSSION THREAD

**Discussion by ssattler** [NREL/ReEDS-2.0#61] [Q&A: Analysis] [answered]
How do I change ReEDS to report all emissions, not just CO2? I tried making some changes to e_report and e_report_dump and I'm running into problems. Thanks!

**Comment by vcelloho** [comment, marked as answer]
*I've edited the response to fix a versioning error in the emit_scale
Within E Reports you want to modify the code that reports CO2 emissions to work for all I'd define new reporting parameters, I've described below an example of how to do this, by creating new indexed on the emissions set e.

You can add the following code to the block of parameters at the start e_reports

emit_all_nat(e,t)                          "--million metric tons CO2-- co2 emissions, national (note: units dependent on emit_scale)"
emit_all_r(e,r,t)                            "--million metric tons CO2-- co2 emissions, regional (note: units dependent on emit_scale)"

The following can be added near where the current CO2 emissions calculations are preformed.

emit_all_r(e,r,t)$[rfeas(r)$tmodel_new(t)] = emit.l(e,r,t) * emit_scale ;
emit_all_nat(e,t)$tmodel_new(t) = sum{r$rfeas(r), emit_all_r(e,r,t) } ; 


Then in e_report_dump you'll need to output the new parameters that you're calculating.

$call 'gdxdump %gdxFileName% format=csv epsout=0 symb=emit_all_nat > %outputsPath%emit_all_nat.csv'
$call 'gdxdump %gdxFileName% format=csv epsout=0 symb=emit_all_r > %outputsPath%emit_all_r.csv'

**Reply by ssattler** [reply]
Thank you so much. 
I am having trouble getting this to write to file, though. The model will complete the solve, but when I open the emit csv files, the files say "GDX file not found: outputs\emit_seq.gdx" Could you help me troubleshoot?

**Reply by ssattler** [reply]
I'm sorry, The model did return an error at 
3096773  emit_all_r(e,r,t)$[rfeas(r)$tmodel_new(t)] = emit.l(e,r,t) * emit_scale(e) ;
****                                                                              $148
**** LINE    608 INPUT       C:\ReEDS_OpenAccess-master\runs\emit_seq\e_report.gms
3096774  emit_all_nat(e,t)$tmodel_new(t) = sum{r$rfeas(r), emit_all_r(e,r,t) } ;

**Reply by wesleyjcole** [reply]
emit_scale is actually a scalar, so you want to drop the (e) from your line of code:

emit_all_r(e,r,t)$[rfeas(r)$tmodel_new(t)] = emit.l(e,r,t) * emit_scale ;

For the forthcoming 2021 version of the model, we have made emit_scale indexed by emission type to improving the scaling in our emission constraints, so using emit_scale(e) will be necessary when using the 2021 version of the model.

**Reply by vcelloho** [reply]
Thanks for catching that issue with my code between versions.

**Reply by ssattler** [reply]
Thanks! It's working great!
